certbot-apache/certbot_apache/tests/augeasnode_test.py

```diff
@@ -144,6 +144,16 @@ def test_set_parameters_delete(self):
                 found = True
         self.assertTrue(found)
 
+    def test_add_child_comment(self):
+        newc = self.config.parser_root.primary.add_child_comment("The content")
+        comments = self.config.parser_root.find_comments("The content")
+        self.assertEqual(len(comments), 1)
+        self.assertEqual(
+            newc.metadata["augeaspath"],
```

Comment:
Should this also be `primary.metadata`?

Reply:
Good catch, and you are right. There's more than meets the eye though: All "read-only" operations towards instance variables already inherently are, as `DualParserNode` returns the value from primary from `__getattr__`.

I'm mainly accessing `primary` directly where I need to count the calls, and in all cases where I need to make "write" operations to not to do the writes twice as you already figured out :)